### PR TITLE
Add complain app

### PR DESCRIPTION
**Pull request by RaiVaibhav**
Add complain app which contain a register form
where user can register there complain.

### CODE REVIEW

File: goodfood/complain/models.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from django.db import models
+
+# Create your models here.
+
+class Post(models.Model):
```
Comment (sks444) [inline goodfood/complain/models.py]: Sorry for not elaborating in the issue;
but a better model name would be `Complain` and form will be `ComplainForm`.

File: goodfood/complain/models.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    #author = models.ForeignKey('auth.User', on_delete=models.CASCADE)
+    name = models.CharField(max_length=200)
+    email = models.EmailField( max_length=100, unique=True)
+    university_name = models.CharField(max_length=200, default='')
+    university_address = models.CharField(max_length=200)
```
Comment (sks444) [inline goodfood/complain/models.py]: use `TextField`(university address can be large)

File: goodfood/complain/models.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    quality_choice = [('1','bad'),('2','bearable'),('3','meidum'),('4','good'),('5','select')]
+    food_quality = models.CharField(max_length=20, choices=quality_choice, default='5')
+    further_comments = models.TextField()
```
Comment (sks444) [inline goodfood/complain/models.py]: further_comments -> complain

File: goodfood/complain/urls.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+from django.urls import re_path
+from . import views
+
+urlpatterns = [
+    re_path('^register/$',views.register, name='register'),
+    re_path(r'^$', views.home, name='home'),
```
Comment (sks444) [inline goodfood/complain/urls.py]: Put it into goodfood/urls.py

File: goodfood/complain/views.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+from django.shortcuts import render, redirect
+from .models import Post
+from complain.forms import PostForm
+
+# Create your views here.
+def home(request):
```
Comment (sks444) [inline goodfood/complain/views.py]: Try to use class-based view

File: goodfood/complain/migrations/0001_initial.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# Generated by Django 2.0.4 on 2018-04-18 16:43
```
Comment (sks444) [inline goodfood/complain/migrations/0001_initial.py]: Delete all the migrations, and do `makemigrations` again we need only one migrations file for a model.

File: goodfood/complain/templates/reg_form.html [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html>
+<body>
+<head>
+<title>Complain Form</title>
+</head>
+<h1>Register your complain</h1>
```
Comment (sks444) [inline goodfood/complain/templates/reg_form.html]: Register -> Submit
And the file name should be `complain_form.html`

File: goodfood/complain/tests.py [hunk 8]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
```
Comment (sks444) [inline goodfood/complain/tests.py]: Remove this if we are not writing tests for now.

File: goodfood/goodfood/urls.py [hunk 9]
```diff
@@ -14,8 +14,9 @@
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
 from django.contrib import admin
-from django.urls import path
+from django.urls import include, re_path
 
 urlpatterns = [
```
Comment (sks444) [inline goodfood/goodfood/urls.py]: Where is home page url?

File: goodfood/goodfood/settings.py [hunk 10]
```diff
@@ -20,7 +20,7 @@
 # See https://docs.djangoproject.com/en/2.0/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'arvtrm45njdf%1d+r(4o_&7u8aqg6n!9i#092-lqnwwrfqip_s'
+SECRET_KEY = 'v&k#+#4p=*&7e(u0x-^9icajgtwdhu0@7^8dgmd8a%%*c8#&k7'
```
Comment (sks444) [inline goodfood/goodfood/settings.py]: don't push the secret_key; it should be secret ;)
we should keep the old secret_key in an env.

File: goodfood/complain/admin.py [hunk 11]
```diff
@@ -0,0 +1,5 @@
+from django.contrib import admin
+from .models import Complain
+# Register your models here.
```
Comment (sks444) [inline goodfood/complain/admin.py]: Remove the comment

File: goodfood/complain/models.py [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+
+# Create your models here.
+
+class Complain(models.Model):
+    #author = models.ForeignKey('auth.User', on_delete=models.CASCADE)
+    name = models.CharField(max_length=200)
+    email = models.EmailField( max_length=100, unique=True)
+    university_name = models.CharField(max_length=200, default='')
+    university_address = models.TextField(max_length=400)
+    quality_choice = [('1','bad'),('2','bearable'),('3','meidum'),('4','good'),('5','select')]
```
Comment (sks444) [inline goodfood/complain/models.py]: Put it at line 6; in the starting

Comment (sks444) [reply goodfood/complain/models.py]: see https://github.com/coala/community/blob/9261c9f6d0777a150e3ee69b146e4d7f52a4eb82/gci/models.py#L55

File: goodfood/complain/models.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+    food_quality = models.CharField(max_length=20, choices=quality_choice, default='5')
```
Comment (sks444) [inline goodfood/complain/models.py]: IntegerField

File: goodfood/complain/views.py [hunk 14]
```diff
@@ -0,0 +1,19 @@
+from django.shortcuts import render, redirect
+from django.contrib import messages
+from .models import Complain
+from complain.forms import ComplainForm
+
+# Create your views here.
+def register(request):
```
Comment (sks444) [inline goodfood/complain/views.py]: `def complain()`

File: goodfood/complain/models.py [hunk 15]
```diff
@@ -0,0 +1,19 @@
+from django.db import models
+
+# Create your models here.
+
+class Complain(models.Model):
+    #author = models.ForeignKey('auth.User', on_delete=models.CASCADE)
```
Comment (sks444) [inline goodfood/complain/models.py]: remove

File: goodfood/complain/models.py [hunk 16]
```diff
@@ -0,0 +1,19 @@
+from django.db import models
+
+# Create your models here.
```
Comment (sks444) [inline goodfood/complain/models.py]: remove

**Comment by RaiVaibhav** [comment]
@sks444  regarding this 
Wait until I add the .coafile and .travis.yml before next push. Will be in an hour.

Use can use https://github.com/RaiVaibhav/python-script/blob/master/.travis.yml these settings which provided by coala to run through docker.
you just have to add the `.coafile`

File: goodfood/complain/models.py (hunk 12 above)
Comment (RaiVaibhav) [reply goodfood/complain/models.py]: If you wanted, I will change it to `intergerfield` 

>Put it at line 6; in the starting
Nope It does't matter

**Comment by RaiVaibhav** [comment]
@CodeVINCI  @sks444  from wait to first get this merge and then push, it causing merge conflict and I don't want to resolve everytime.

File: goodfood/complain/models.py (hunk 12 above)
Comment (sks444) [reply goodfood/complain/models.py]: >It does't matter

I know :)
But taht way it looks good and symmetrical.

**Comment by sks444** [comment]
@RaiVaibhav,  yeah sorry;
This was the last time, now i am done with base setup :)

**Comment by RaiVaibhav** [comment]
I am closing this PR for now because if `.coafile` I have to solve a ton of conflicts which I don't want .

File: goodfood/goodfood/urls.py (hunk 9 above)
Comment (RaiVaibhav) [reply goodfood/goodfood/urls.py]: I have not added that if anyone wants he/she is welcome I, my work was to add a working `complain` app and it is working.

**Pull request closed without merging**